pkg/collector/autodiscovery/autoconfig.go

```diff
@@ -174,7 +176,19 @@ func (ac *AutoConfig) GetChecksByName(checkName string) []check.Check {
 func (ac *AutoConfig) getAllConfigs() []check.Config {
 	configs := []check.Config{}
 	for _, pd := range ac.providers {
-		cfgs, _ := ac.collect(pd)
```

Comment:
I'm afraid this breaks Autodiscovery, `collect` does other things beside invoking `Collect()` on any provider